webapp/channels/src/packages/mattermost-redux/src/selectors/entities/posts.ts

```diff
@@ -269,6 +269,59 @@ function formatPostInChannel(post: Post, previousPost: Post | undefined | null,
     };
 }
 
+export function getLatestPostIdToReact(state: GlobalState, channelId: string, rootId = '') {
+    const postsIds = rootId ? getPostsInThread(state)[rootId] : getPostIdsInChannel(state, channelId);
+    if (!postsIds) {
+        return '';
+    }
+
+    const allPosts = getAllPosts(state);
+
+    for (const postId of postsIds) {
+        if (isCombinedUserActivityPost(postId)) {
+            continue;
+        }
+
+        const post = allPosts[postId];
+        if (!post) {
+            continue;
+        }
+
+        if (post.delete_at) {
+            continue;
+        }
+
+        return postId;
+    }
+
+    if (rootId && allPosts[rootId] && !allPosts[rootId].delete_at) {
+        return rootId;
+    }
+
+    return '';
+}
+
+export function getLatestPostToEdit(state: GlobalState, channelId: string, rootId = '') {
```

Comment:
Same question here regarding `getLatestReplyablePostId`

Reply:
I'm also wondering why `getLatestPostIdToReact` and `getLatestPostToEdit` are different from one another since the old version of the post textbox uses the result of `getLatestReplyablePostId` for both of those

Reply:
I can take another look. My idea is that a post to edit, a post to react and a post to reply are slightly different, and make sense they are different, but maybe there is some overlap we can simplify. 0/5

Reply:
I refactored `getLatestReplyablePostId` to use `getLastestPodIdToReact` (renamed into `getLastInteractablePostId`).